src/View/Helper/LinkHelper.php

```diff
@@ -76,17 +76,34 @@ public function fromAPI($apiUrl): void
      * Returns sort url by field direction
      *
      * @param string $field the Field.
+     * @param bool $resetPage flag to reset pagination.
      * @return string
      */
-    public function sortUrl($field): string
+    public function sortUrl($field, $resetPage = true): string
     {
-        $sort = $this->getView()->getRequest()->getQuery('sort');
+        $replace = [];
+        $request = $this->getView()->getRequest();
+        $sort = $request->getQuery('sort');
         $sortValue = $field; // <= ascendant order
         if ($sort === $field) { // it was ascendant sort
             $sortValue = '-' . $field; // <= descendant order
         }
+        $replace['sort'] = $sortValue;
 
-        return $this->replaceParamUrl('sort', $sortValue);
+        $currentPage = Hash::get($request->getQueryParams(), 'page');
+        if (isset($currentPage)) {
+            if ($resetPage) {
+                $replace['page'] = 1;
+            } else {
+                // reset the page if sort field has changed
+                parse_str(parse_url(urldecode($request->referer()), PHP_URL_QUERY), $prevQuery);
+                if (Hash::get($prevQuery, 'sort') !== $sortValue) {
+                    $replace['page'] = 1;
+                }
+            }
+        }
+
+        return $this->replaceQueryParams($replace);
     }
```

Comment:
it could be "one level" block, like following:

```
if (!isset($currentPage)) {
   $this->replaceQueryParams($replace);
}

if ($resetPage) {
    $replace['page'] = 1;

    return $this->replaceQueryParams($replace);
}

// reset the page if sort field has changed
parse_str(parse_url(urldecode($request->referer()), PHP_URL_QUERY), $prevQuery);
if (Hash::get($prevQuery, 'sort') !== $sortValue) {
    $replace['page'] = 1;
}

return $this->replaceQueryParams($replace);
```